migrations/3_deploy_public_sale_contracts.js

```diff
@@ -0,0 +1,96 @@
+// The Carry token and the tokensale contracts
+// Copyright (C) 2018 Carry Protocol
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+const CarryToken = artifacts.require("CarryToken");
+const CarryPublicTokenCrowdsale =
+    artifacts.require("CarryPublicTokenCrowdsale");
+
+function timestamp(iso8601) {
+    return +new Date(iso8601) / 1000 >> 0;
+}
+
+const publicSale = {
+    // See also <https://carryprotocol.io/#section-token-distribution>.
+    //   1 ETH = 65,000 CRE
+    //   1 wei = 6.5E-14 CRE
+    // Note that both currencies have the same precision on
+    // their minor units: 18 decimals.
+    rate: 65000,
+
+    // Max cap: 5000.41 ETH = 373,781,000 CRE
+    cap: web3.toWei(5000410, "finney"),
```

Comment:
Just to check, is cap of 5000.41 ETH correct?

Reply:
@qria It's not determined yet; probably changed in the future.